### ISSUE TITLE
Unable to run any tasks on a remote windows machine using openssh

### ISSUE THREAD

**Issue by rjawiygvozd** [microsoft/vscode-remote-release#5989]
I'm connecting to a Windows 10 machine from a Linux host. Connection is successful, project is opened, terminal is working properly.
I'm using powershell 7 installed in Program Files.

However, if I launch any task, I'm getting the following error:

> The terminal process failed to launch: A native exception occurred during launch (Unable to start terminal process: CreateProcess failed).

So far I tried following:
1) changing the _terminal.integrated.automationShell.windows_ setting. I tried "cmd", "pwsh", "powershell" and the full path to git bash, same result. I can tell the setting is applied because I can get "not found" error if I set the wrong path.
2) setting default openssh shell to the same pwsh.exe that is used for tasks and terminal, didn't help.

I also tried changing _terminal.integrated.profiles.windows_ and _terminal.integrated.defaultProfile.windows_ but apparently they're irrelevant here.

Edit: I tried tasks with "shell", "process" and "gulp" types, none worked

**Comment by tanhakabir** [comment]
@meganrogge or @Tyriar do you guys have an idea of why this error happened?

**Comment by meganrogge** [comment]
which version are you on? did this start happening recently?

cc @alexr00

**Comment by meganrogge** [comment]
I see `CreateProcess failed` is a winpty error

**Comment by meganrogge** [comment]
Can you share your settings.json file and also do a trace log while creating a terminal? https://github.com/microsoft/vscode/wiki/Terminal-Issues#enabling-trace-logging

**Comment by rjawiygvozd** [comment]
I'm trying this for the first time so I don't have experience of this working.
Versions I can think of:
vscode 1.62.3
remote - ssh 0.66.1
current endeavorOS/arch linux
windows 10 21h2
powershell 7.2

openssh installed recently using 
> Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0

this seems to be the log part
> [2021-12-03 04:44:36.758] [renderer1] [trace] KeybindingService#dispatch Alt+T undefined
> [2021-12-03 04:44:36.775] [renderer1] [trace] Layout#layout, height: 994, width: 1920
> [2021-12-03 04:44:37.801] [renderer1] [trace] CommandService#executeCommand workbench.action.tasks.runTask
> [2021-12-03 04:44:37.802] [renderer1] [trace] onWillActivateByEvent:  onCommand:workbench.action.tasks.runTask
> [2021-12-03 04:44:37.802] [renderer1] [trace] Layout#layout, height: 1019, width: 1920
> [2021-12-03 04:44:38.090] [renderer1] [trace] CommandService#executeCommand setContext
> [2021-12-03 04:44:43.785] [renderer1] [trace] terminalInstance#ctor (instanceId: 5) {"name":"build (tumarket)","description":"Task","executable":"pwsh","args":"/d /c msbuild ./src/TuMarket/TuMarket.Web/Web.csproj /t:build /v:m","waitOnExit":"Terminal will be reused by tasks, press any key to close it.","initialText":"\u001b[1m> Executing task in folder tumarket: msbuild ./src/TuMarket/TuMarket.Web/Web.csproj /t:build /v:m <\u001b[0m\n","cwd":{"$mid":1,"path":"/\\work\\projects\\tumarket","scheme":"vscode-remote","authority":"ssh-remote+win"},"isFeatureTerminal":true,"useShellEnvironment":true,"icon":{"id":"tools"}}
> [2021-12-03 04:44:43.878] [renderer1] [trace] Stopped polling vscode-remote://ssh-remote%2Bwin/c%3A/Users/oleg/.vscode-server/data/logs/20211203T044130/exthost2/output_logging_20211203T044333/5-Razor%20Log.log
> [2021-12-03 04:44:44.415] [renderer1] [trace] CommandService#executeCommand setContext
> [2021-12-03 04:44:44.468] [renderer1] [trace] Spawning remote agent process {"terminalId":9}
> [2021-12-03 04:44:46.858] [renderer1] [debug] Terminal process exit (instanceId: 5) with code undefined
> [2021-12-03 04:44:46.858] [renderer1] [debug] Terminal process exit (instanceId: 5) state 3
> [2021-12-03 04:44:50.151] [renderer1] [trace] KeybindingService#dispatch Space undefined
> [2021-12-03 04:44:50.152] [renderer1] [trace] terminalInstance#dispose (instanceId: 5)
> [2021-12-03 04:44:50.425] [renderer1] [trace] CommandService#executeCommand setContext
> 

this is the settings.json

```
{
  "mssql.connections": [
    {
      //skipped
    }
  ],
  "editor.minimap.enabled": false,
  "window.menuBarVisibility": "toggle",
  "terminal.integrated.profiles.windows": {
    "PowerShell -NoProfile": {
      "source": "PowerShell",
      "args": [
        "-NoProfile"
      ]
    },
    "pwsh1": {
      "path": "pwsh.exe",
      "args": []
    }
  },
  "terminal.integrated.defaultProfile.windows": "PowerShell",
  "terminal.integrated.automationShell.windows": "pwsh"
}
```

**Comment by rjawiygvozd** [comment]
I also tried adding
> "terminal.integrated.windowsEnableConpty": true
nothing changed

**Comment by meganrogge** [comment]
thanks for that info. Looks like the process was killed during launch, likely as a result of badshell args.

Looks like your shell args are "/d /c msbuild ./src/TuMarket/TuMarket.Web/Web.csproj /t:build /v:m"`

Which task are you trying to run?

**Comment by rjawiygvozd** [comment]
This is a task that builds a .net 4.7 project. It works when I normally run it on windows.
I also tried running a gulp task and a task that only runs a powershell script and they cause the same error.

**Comment by rjawiygvozd** [comment]
btw "/d /c" part is something appended automatically, i imagine that's always the case

these are the tasks
```

{
    "label": "build",
    "type": "shell",
    "command": "msbuild",
    "args": [
        "./src/TuMarket/TuMarket.Web/Web.csproj",
        "/t:build",
        "/v:m"
    ],
    "group": "build",
    "presentation": {
        "reveal": "always"
    },
    "problemMatcher": "$msCompile"
},
{
    "label": "setPortableDebugType",
    "command": "powershell",
    "args": [
        ".vscode/setPortableDebugType.ps1"
    ],
    "type": "shell",
    "problemMatcher": []
},
{
    "type": "gulp",
    "task": "watch-tumarket-bundle",
    "label": "gulp: watch-tumarket-bundle",
    "problemMatcher": [
        "$gulp-tsc"
    ]
}
```

**Comment by rzhao271** [comment]
@rjawiygvozd is the issue fixed for you on the latest Insiders (https://code.visualstudio.com/insiders/)?

**Comment by rjawiygvozd** [comment]
yes, it's working now

**Comment by rjawiygvozd** [comment]
there seems to also be an issue of a task not launching if "command" contains whitespaces which I was able to avoid by adding a folder with the executable I needed to PATH so I don't have to use a full path for it. I didn't have to do that on windows locally.

this link is related to the same problem apparently, at least I was getting the same error and the reason was the same
https://stackoverflow.com/questions/47638737/powershell-you-must-provide-a-value-expression-following-the-operator-erro

**Comment by alexr00** [comment]
Marking as verified because of https://github.com/microsoft/vscode-remote-release/issues/5989#issuecomment-987563321, thank you @rjawiygvozd.

Can you share an example that causes the other error you're seeing?

**Comment by rjawiygvozd** [comment]
The problem was with this task

```
{
    "label": "launch",
    "type": "shell",
    "command": "c:/program files/iis express/iisexpress",
    //"command": "iisexpress",
    "args": [
        "/site:TuMarket"
        //"/path:${workspaceFolder}\\src\\TuMarket\\TuMarket.Web",
        //"/port:8081"
    ],
    "presentation": {
        "reveal": "silent"
    },
    "problemMatcher": [
        {
            "pattern": [
                {
                    "regexp": ".",
                    "file": 1,
                    "location": 2,
                    "message": 3
                }
            ],
            "background": {
                "activeOnStart": true,
                "beginsPattern": ".",
                "endsPattern": "^IIS Express is running."
            }
        }
    ],
    "dependsOn": [
        "setPortableDebugType",
        "build"
    ],
    "dependsOrder": "sequence",
    "isBackground": true
},
```

The error was

> You must provide a value expression following the '/' operator

referring specifically to "/site" in the first argument

It was solved when I replaced full path to just "iisexpress" as in commented line

**Comment by alexr00** [comment]
Thanks @rjawiygvozd, same problem, different location. I've pushed a fix, but it won't be available in insiders until after we release 1.63 (probably at the end of the week).
